week2/ShoppingList/index.js

```diff
@@ -0,0 +1,189 @@
+import { ITEMLIST } from "./ITEMLIST.js";
+
+const ITEM_LIST_KEY = "itemList";
+
+//console.dir을 이용하여 객체의 더 자세한 속성들 확인
+const allSection = document.querySelector("#all");
+const albumSection = document.querySelector('#album');
+const movieSection = document.querySelector('#movie');
+const travelSection = document.querySelector('#travel');
+
+const mainContainer = document.querySelector("#mainContainer");
+
+// 매번 렌더링시마다 filter해서 요소들을 걸러내는게 비효율적이라 판단.
+// 따라서 미리 필터링 된 요소들을 저장해 둠.
+// 참고로, const로 선언한 배열 자체는 수정(대체)이 불가능하지만,
+// 배열의 '요소'들은 수정이 가능하다.
+let allObjects;
+const storageObjects = localStorage.getItem(ITEM_LIST_KEY);
+if(storageObjects !== null) //로컬 스토리지에서 가져온 값으로 매번 갱신해줘야함. (아니면 계속 덮어쓰기 되는 버그 발생)
+{
+  allObjects = JSON.parse(storageObjects);
+}
+else{
+  allObjects = ITEMLIST;
+}
+const albumObjects = ITEMLIST.filter((obj) => obj.category === "album"); 
+const movieObjects = ITEMLIST.filter((obj) => obj.category === "movie"); 
+const travelObjects = ITEMLIST.filter((obj) => obj.category === "travel"); 
+
+function removeOtherSection(){
+  const all = document.getElementById("all");
+  const album = document.getElementById("album");
+  const movie = document.getElementById("movie");
+  const travel = document.getElementById("travel");
+
+  if(all !== null) //해당 섹션이 존재한다면,
+  {
+    all.remove(); //해당 섹션 요소 삭제
+    console.log("all 섹션 삭제됨");
+  }
+  else if(album !== null) //else if를 사용한 이유 -> 어차피 하나의 섹션만 렌더링되어 있을것이므로!
```

Comment:
```suggestion
  else if(album) 
```
album이 존재할 때의 경우를 조건부 렌더링하고싶은 거라면, 간단하게 album만 넣어줘도 됩니다!